### ISSUE TITLE
[BUG] statsforecast 1.6.0 deprecates parallel argument

### ISSUE THREAD

**Issue by ngupta23** [sktime/sktime#5207]
**Describe the bug**

In nixtla statsforecast version 1.6.0, the parallel argument has been deprecated in AutoARIMA. This causes the sktime wrapper to fail.

https://github.com/Nixtla/statsforecast/commit/478e774c03df1b7913520d44f5e4d3eaef2e2e13

**To Reproduce**

```python
!pip install sktime
!pip install statsforecast==1.6.0

from sktime.datasets import load_airline
from sktime.forecasting.statsforecast import StatsForecastAutoARIMA
y = load_airline()
forecaster = StatsForecastAutoARIMA(  
    sp=12, d=0, max_p=2, max_q=2
)
forecaster.fit(y)  
y_pred = forecaster.predict(fh=[1,2,3])  
```
![image](https://github.com/sktime/sktime/assets/33585645/df7dfc46-6a26-40ba-9b0e-39184262a748)

```python
!pip install statsforecast==1.5.0

forecaster = StatsForecastAutoARIMA(  
    sp=12, d=0, max_p=2, max_q=2
)
forecaster.fit(y)  
y_pred = forecaster.predict(fh=[1,2,3]) 
```

This works

**Expected behavior**
Should work with latest version of statsforecast

**Additional context**

First found in pycaret unit tests

![image](https://github.com/sktime/sktime/assets/33585645/3f650d02-73bc-4356-8724-194d196b4dbd)

**Versions**

latest install - 0.22.0

**Comment by yarnabrina** [comment]
Duplicate of #5150

@fkiraly added a bound for now in `main`, and I believe we are waiting for comments from Nixtla as they interfaced `AutoARIMA`.

**Comment by fkiraly** [comment]
Oh, we must have miscommunicated then!
I thought we were going to fix it as part of the adapter, around the work that you (@yarnabrina) and @arnaujc91 were doing in https://github.com/sktime/sktime/pull/5112?

If that was not the plan, I would suggest to make a small utility that allows to pass parameters based on kwargs being present in the init.

**Comment by fkiraly** [comment]
I thought this will fix it:
https://github.com/sktime/sktime/pull/5112

because the problem with older `statsforecast` versions is the same as the problem with newer versions - they are constantly changing their init args...

**Comment by yarnabrina** [comment]
@fkiraly I think I didn't make myself clear.

Let's say current `sktime` estimator supports `S1` set of arguments which is in sync. Over few next releases, `statsforecast` shifts to `S2`, `S3` etc. All these have significant intersections,  but each have their unique elements, but as of now `sktime` is stuck with `S1` still.

My understanding in @arnaujc91's PR will ensure unique `S1` arguments will continue to work as we'll not pass them internally, but they won't allow unique ones of `S2`, `S3`, etc. and users will loose ability to use latest  `statsforecast` features.

Should we not add support for new arguments, as and when added, and remove supportfrom old ones, after some decided time? That'd mean change in concrete estimator class written by Nixtla, so I thought we are waiting for them

(And this problem is theoretically valid for all estimators I suppose, not just `statsforecast`.)

Does it make sense? If I'm missing something and this is already handled by #5112, please let me know.

**Comment by fkiraly** [comment]
> Should we not add support for new arguments, as and when added, and remove supportfrom old ones, after some decided time? That'd mean change in concrete estimator class written by Nixtla, so I thought we are waiting for them

Yes, if we could although I don't quite see how, technically, the estimator would "self-update" to new arguments that Nixtla add.

At least, I think, we should support all arguments that have appeared and possibly disappeared at some point over the version support period.

There is further the requirement of "not breaking downstream code" which implies deprecation, as well as not changing the order of arguments without a change/deprecation loop; and changes to the full set of arguments can be made without deprecation procedure only by appending arguments at the end.

> If I'm missing something and this is already handled by https://github.com/sktime/sktime/pull/5112, please let me know.

I think this is handled by #5112 in the sense that it minimizes maintenance subject to the above requirements - we simply add new arguments, as and when they appear, to the `__init__` of the interface - the internal machinery does the rest.

This is the minimal maintenance level imo that we can achieve without dynamically generating class signatures, which is a hard and risky topic.
